### PR TITLE
Make IMemory interface follow the convention instead of returning tuple

### PR DESCRIPTION
**Pull request by Danieladu**
fix: #3162

### CODE REVIEW

File: libraries/Microsoft.Bot.Expressions/Memory/IMemory.cs [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
+        object SetValue(string path, object value);
+
         /// <summary>
         /// Get value from a given path, it can be a simple indenfiter like "a", or
         /// a combined path like "a.b", "a.b[2]", "a.b[2].c", inside [] is guranteed to be a int number or a string.
         /// </summary>
         /// <param name="path">memory path.</param>
         /// <returns> resovled value and error messsage if any. </returns>
-        (object value, string error) GetValue(string path);
+        (object value, string error) TryGetValue(string path);
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Expressions/Memory/IMemory.cs]: I think we should return boolean, because that's what "convention" means.

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.AspNet.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.AspNet.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Testing.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Testing/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Streaming.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Streaming/4.6.3)

File: libraries/Microsoft.Bot.Expressions/Memory/MemoryBase.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
+
+namespace Microsoft.Bot.Expressions.Memory
+{
+    public abstract class MemoryBase : IMemory
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Expressions/Memory/MemoryBase.cs]: I'm not sure about the convention here, it's MemoryBase or BaseMemory?

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.AspNet.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.AspNet.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Testing.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Testing/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Streaming.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Streaming/4.6.3)

**Comment by fuselabs** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.Luis.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.Luis/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.AI.QnA.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.AI.QnA/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.ApplicationInsights.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.ApplicationInsights/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Azure.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Dialogs.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Dialogs/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.ApplicationInsights.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Integration.AspNet.Core.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Integration.AspNet.Core/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.TemplateManager.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.TemplateManager/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Builder.Testing.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Builder.Testing/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Configuration.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Configuration/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Connector.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Connector/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Schema.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Schema/4.6.3)
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.Bot.Streaming.dll** compared against [version 4.6.3](https://www.nuget.org/packages/Microsoft.Bot.Streaming/4.6.3)

**Pull request merged**